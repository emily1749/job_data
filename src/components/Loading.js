import './Loading.css';
import React from 'react';
import { connect } from 'react-redux';

const Loading = ({ message, loading }) => {
  return (
    <div>
      {loading ? (
        <div className='lds-ellipsis'>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
        </div>
      ) : (
        <div className='info'>
          <p>{message}</p>
        </div>
      )}
    </div>
  );
};

const mapStateToProps = state => {
  return {
    message: state.jobData.message,
    loading: state.jobData.loading,
  };
};

export default connect(mapStateToProps)(Loading);
