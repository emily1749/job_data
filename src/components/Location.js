import React from 'react';
import { connect } from 'react-redux';

import {
  setBubbleColor,
  setQuickColor,
  setMergeColor,
  setCityLocation,
  setStateLocation,
  fetchJobData,
} from '../actions';

const Location = ({
  onSort,
  cityLocation,
  stateLocation,
  buttonColor,
  setBubbleColor,
  setMergeColor,
  setQuickColor,
  setCityLocation,
  setStateLocation,
  fetchJobData,
}) => {
  let city;
  let state;

  const onCityInputChange = e => setCityLocation(e.target.value);

  const onStateInputChange = e => setStateLocation(e.target.value);

  const onLocationSubmit = e => {
    e.preventDefault();

    if (onSort === false) {
      let city = cityLocation;
      let state = stateLocation;

      if (city && state) {
        console.log('in here');
        city = city.replace(' ', '+');
        setMergeColor('');
        setBubbleColor('');
        setQuickColor('');

        const fetchData = async () => {
          await fetchJobData(city, state);
        };
        fetchData();
      }
    }
  };

  return (
    <div>
      <div>
        <h2>Location</h2>
      </div>

      <form onSubmit={onLocationSubmit}>
        <div className='location-container'>
          <div>
            <label>City:</label>

            <input
              type='text'
              name='city'
              className='input-text'
              value={city}
              onChange={onCityInputChange}
            />
          </div>

          <div>
            <label>State:</label>

            <input
              type='text'
              name='state'
              className='input-text'
              value={state}
              onChange={onStateInputChange}
            />
          </div>

          <div>
            <div className='buttonHolder'>
              <button className='btn' style={{ color: buttonColor }}>
                Submit Location
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    onSort: state.onSort,
    cityLocation: state.cityLocation,
    stateLocation: state.stateLocation,
    buttonColor: state.buttonColor,
  };
};

const mapDispatchToProps = {
  setBubbleColor,
  setMergeColor,
  setQuickColor,
  setCityLocation,
  setStateLocation,
  fetchJobData,
};

export default connect(mapStateToProps, mapDispatchToProps)(Location);
