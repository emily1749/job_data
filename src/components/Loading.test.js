import React from 'react';
import Loading from './Loading';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { mount } from 'enzyme';
import sinon from 'sinon';
import thunk from 'redux-thunk';
import Location from './Location';
import moxios from 'moxios';

const initialState = {
  jobData: {
    jobDataCopy: [],
    loading: false,
    error: false,
    locationSubmitted: false,
    message: 'Please enter location',
  },
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState);

describe('App Component', () => {
  let props;

  beforeEach(() => {
    props = {
      jobData: {
        jobDataCopy: [],
        loading: false,
        error: false,
        locationSubmitted: false,
        message: 'Please enter location',
      },
    };
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Loading {...props} />
      </Provider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
// import React from 'react';
// import { shallow } from 'enzyme';
// import { mount } from 'enzyme';
// // import sinon from 'sinon'
// import thunk from 'redux-thunk';
// import Loading from './Loading';

// // import ConnectedUserInput, { UserInput } from '../src/components/UserInput'
// import * as actionCreators from '../actions';
// import { Provider } from 'react-redux';
// import configureStore from 'redux-mock-store';

// const middlewares = [thunk];
// const mockStore = configureStore(middlewares);
// const initialState = {
//   jobData: {
//     jobDataCopy: [],
//     loading: false,
//     error: false,
//     locationSubmitted: false,
//     message: 'Please enter location',
//   },
// };
// const store = mockStore(initialState);

// const wrapper = mount(
//   <Provider store={store}>
//     <Loading />
//   </Provider>
// );

// console.log(wrapper.debug());

// const testTechInputInAllTec hs = 'css';

// test('has access to "Loading" state', () => {
//   wrapper.setProps();
//   // expect(onSortProp).toBe(onSort);
// });

// describe('(Component) UserInput', () => {
//   it('renders...', () => {
//     expect(wrapper).to.have.length(1);
//   });

//   it('should have message if loading is false', () => {
//     moxios.wait(() => {
//       const request = moxios.requests.mostRecent();
//       request.respondWith({
//         status: 200,
//         response: 'testResponse',
//       });
//     });

//     const res = store.dispatch(actionCreators.fetchJobData('city', 'state'));
//     expect(store.getState().jobData.locationSubmitted).equals(true);
//     // .then(() => {
//     //   const newState = store.getState();
//     //   // expect(newState.secretWord).toBe('test');
//     // });
//   });

//   it('should not have message display if loading is true', () => {});
// });
