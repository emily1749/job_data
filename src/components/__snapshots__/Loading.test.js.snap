// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<div>
  <div
    className="info"
  >
    <p>
      Please enter location
    </p>
  </div>
</div>
`;
